Updating OpenAI conversation component to include custom endpoint support (#4678)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-conversation/openai.md

|--------------------|:--------:|---------|---------|
| `key`   | Y | API key for OpenAI. | `mykey` |
| `model` | N | The OpenAI LLM to use. Defaults to `gpt-4-turbo`.  | `gpt-4-turbo` |

> | `endpoint` | N | Custom API endpoint URL for OpenAI API-compatible services. If not specified, the default OpenAI API endpoint will be used. | `https://api.openai.com/v1` |

Reviewer: ```suggestion
| `endpoint` | N | Custom API endpoint URL for OpenAI API-compatible services. If not specified, the default OpenAI API endpoint is used. | `https://api.openai.com/v1` |
```
